nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogic.java

```diff
@@ -339,6 +372,24 @@ protected void addWriteEvent(WriteBatcher writeBatcher, WriteEvent writeEvent) {
 
     protected WriteEvent buildWriteEvent(ProcessContext context, ProcessSession session, FlowFile flowFile) {
         String uri = flowFile.getAttribute(context.getProperty(URI_ATTRIBUTE_NAME).getValue());
+        uri = applyPropertiesToUri(context, flowFile, uri);
+
+        DocumentMetadataHandle metadata = buildMetadataHandle(context, flowFile, context.getProperty(COLLECTIONS), context.getProperty(PERMISSIONS));
+        final byte[] content = new byte[(int) flowFile.getSize()];
+        session.read(flowFile, inputStream -> StreamUtils.fillBuffer(inputStream, content));
+
+        BytesHandle handle = new BytesHandle(content);
+        applyPropertiesToContentHandle(context, uri, handle);
+
+        registerFlowFileInMap(flowFile, session);
+
+        return new WriteEventImpl()
```

Comment:
I'm a little concerned that the NiFi implementation is reaching into the DMSDK implementation, but let's not pursue that now.